src/dependencies/dss-direct-deposit/D3MInit.sol

```diff
@@ -0,0 +1,326 @@
+            dss.jug.init(ilk);
+        }
+        dss.vat.file(ilk, "line", cfg.gap);
+        dss.vat.file("Line", dss.vat.Line() + cfg.gap - previousIlkLine);
+        DssAutoLineAbstract(dss.chainlog.getAddress("MCD_IAM_AUTO_LINE")).setIlk(
+            ilk,
+            cfg.maxLine,
+            cfg.gap,
+            cfg.ttl
+        );
+        dss.spotter.poke(ilk);
+
+        GemAbstract gem = GemAbstract(ID3MPool(d3m.pool).redeemable());
+        IlkRegistryAbstract(dss.chainlog.getAddress("ILK_REGISTRY")).put(
+            ilk,
+            address(hub),
+            address(gem),
+            gem.decimals(),
+            4,
+            address(oracle),
+            address(0),
+            gem.name(),
+            gem.symbol()
+        );
+
+        string memory clPrefix = ScriptTools.ilkToChainlogFormat(ilk);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_POOL"))), d3m.pool);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_PLAN"))), d3m.plan);
+        dss.chainlog.setAddress(ScriptTools.stringToBytes32(string(abi.encodePacked(clPrefix, "_ORACLE"))), d3m.oracle);
+    }
+
+    /**
+     * @dev This works for V2 and V3 pool adapters.
+     */
+    function initAavePool(
+        DssInstance memory dss,
+        D3MInstance memory d3m,
+        D3MCommonConfig memory cfg,
+        D3MAavePoolConfig memory aaveCfg
+    ) internal {
+        D3MAavePoolLike pool = D3MAavePoolLike(d3m.pool);
+
+        // Sanity checks
+        require(pool.hub() == cfg.hub, "Pool hub mismatch");
+        require(pool.ilk() == cfg.ilk, "Pool ilk mismatch");
+        require(pool.vat() == address(dss.vat), "Pool vat mismatch");
+        require(pool.dai() == address(dss.dai), "Pool dai mismatch");
+        require(pool.adai() == aaveCfg.adai, "Pool adai mismatch");
+        require(pool.stableDebt() == aaveCfg.stableDebt, "Pool stableDebt mismatch");
+        require(pool.variableDebt() == aaveCfg.variableDebt, "Pool variableDebt mismatch");
+
+        pool.file("king", aaveCfg.king);
+    }
+
+    function initCompoundPool(
+        DssInstance memory dss,
+        D3MInstance memory d3m,
+        D3MCommonConfig memory cfg,
+        D3MCompoundPoolConfig memory compoundCfg
+    ) internal {
+        D3MCompoundPoolLike pool = D3MCompoundPoolLike(d3m.pool);
+
+        // Sanity checks
+        require(pool.hub() == cfg.hub, "Pool hub mismatch");
+        require(pool.ilk() == cfg.ilk, "Pool ilk mismatch");
+        require(pool.vat() == address(dss.vat), "Pool vat mismatch");
+        require(pool.dai() == address(dss.dai), "Pool dai mismatch");
+        require(pool.comptroller() == compoundCfg.comptroller, "Pool comptroller mismatch");
+        require(pool.comp() == compoundCfg.comp, "Pool comp mismatch");
+        require(pool.cDai() == compoundCfg.cdai, "Pool cDai mismatch");
+
+        pool.file("king", compoundCfg.king);
+    }
```

Comment:
This function apparently is not used. Could potentially be removed.

Reply:
Same as above.